src/cli/Peers.ts

```
@@ -0,0 +1,387 @@
+          successful = await client.addPeer(nodePath, publicKey, connectedAddress?.toString() ?? '', relayPublicKey);
+        } else {
+          const publicKeyPath = options.publicKey;
+          // read in publicKey if it exists
+          const publicKey = fs.readFileSync(publicKeyPath).toString();
+
+          const connectedAddress = options.connectedAddress;
+          const relayPublicKeyPath = options.relayKey;
+          const relayPublicKey = fs.readFileSync(relayPublicKeyPath).toString();
+
+          successful = await client.addPeer(nodePath, publicKey, connectedAddress, relayPublicKey);
+        }
+
+        if (successful) {
+          pkLogger('peer successfully added to peer store', PKMessageType.SUCCESS);
+        } else {
+          pkLogger('something went wrong, peer was not added to peer store', PKMessageType.WARNING);
+        }
+      }),
+    );
+}
+
+function makeUpdatePeerInfoCommand() {
+  return new commander.Command('update')
+    .description('update the peer info for a particular public key')
+    .option('--node-path <nodePath>', 'node path')
+    .option('-c, --current-node', 'only list the peer information for the current node, useful for sharing')
+    .option('-p, --public-key <publicKey>', 'path to the file which contains the public key')
+    .option('-b, --base64 <base64>', 'decode the peer info from a base64 string')
+    .option('-ch, --connected-host <connectedHost>', 'update the connected addresss host')
+    .option('-cp, --connected-port <connectedPort>', 'update the connected addresss port')
+    .option('-r, --relay-key <relayKey>', 'path to the file which contains the public key of the relay peer')
+    .option('-v, --verbose', 'increase verbosity level by one')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const status = await client.getAgentStatus();
```

If you had the class `Agent`, then the status can be part of its construction. Plus what happens when there's a race condition? Obviously we cannot be checking status all the time. Instead you need to throw an exception if the status is not online. Catch this exception whenever you are sending requests to the agent.

Then a possible handler is to restart the agent. But of course this depends as well. As we want to make it work in situations where agent lifecycle is managed externally. And starting agent is always assuming we are running agent locally and we are happy to daemonize with lifecycle handler being some pid lockfile somewhere. This idea is only for ease of use and shouldn't always be used.

Also the `pk-agent` should always be running as foreground.

Users should generally not be running `pk-agent` at all. Instead they are meant to be running `pk` and then optionally start the agent if it isn't started.